### ISSUE TITLE
aeson:1.0.2.1 not working in eta

### ISSUE THREAD

**Issue by politrons** [typelead/eta#831]
With the current version does not work when I add this dependency

`compile eta('aeson:1.0.2.1')
`
I´m having this error

```
> Task :setupEnvironmentEta UP-TO-DATE
etlas: Could not resolve dependencies:
trying: Eta-Haskell-server-0.0.0 (user goal)
trying: directory-1.3.1.0/installed-AeQ... (dependency of
eta-boot-0.8.0/installed-Bq9...)
next goal: aeson (dependency of Eta-Haskell-server-0.0.0)
rejecting: aeson-1.2.3.0, aeson-1.2.2.0, aeson-1.2.1.0, aeson-1.2.0.0,
aeson-1.1.2.0, aeson-1.1.1.0, aeson-1.1.0.0 (conflict: Eta-Haskell-server =>
aeson==1.0.2.1)
rejecting: aeson-1.0.2.1 (conflict: directory =>
time==1.8.0.3/installed-WEw..., aeson => time>=1.1.1.4 && <1.7)
rejecting: aeson-1.0.2.0, aeson-1.0.1.0, aeson-1.0.0.0, aeson-0.11.3.0,
aeson-0.11.2.1, aeson-0.11.2.0, aeson-0.11.1.4, aeson-0.11.1.3,
aeson-0.11.1.2, aeson-0.11.1.1, aeson-0.11.1.0, aeson-0.11.0.0, aeson-0.9.0.1,
aeson-0.9.0.0, aeson-0.8.1.1, aeson-0.8.1.0, aeson-0.8.0.2, aeson-0.7.0.6,
aeson-0.7.0.4, aeson-0.6.2.1, aeson-0.6.2.0, aeson-0.6.1.0, aeson-0.6.0.2,
aeson-0.6.0.1, aeson-0.6.0.0, aeson-0.5.0.0, aeson-0.4.0.1, aeson-0.4.0.0,
aeson-0.3.2.14, aeson-0.3.2.13, aeson-0.3.2.12, aeson-0.3.2.11,
aeson-0.3.2.10, aeson-0.3.2.9, aeson-0.3.2.8, aeson-0.3.2.7, aeson-0.3.2.6,
aeson-0.3.2.5, aeson-0.3.2.4, aeson-0.3.2.3, aeson-0.3.2.2, aeson-0.3.2.1,
aeson-0.3.2.0, aeson-0.3.1.1, aeson-0.3.1.0, aeson-0.3.0.0, aeson-0.2.0.0,
aeson-0.1.0.0, aeson-1.4.0.0, aeson-1.3.1.1, aeson-1.3.1.0, aeson-1.3.0.0,
aeson-1.2.4.0, aeson-0.10.0.0, aeson-0.8.0.1, aeson-0.8.0.0, aeson-0.7.0.5,
aeson-0.7.0.3, aeson-0.7.0.2, aeson-0.7.0.1, aeson-0.7.0.0 (conflict:
Eta-Haskell-server => aeson==1.0.2.1)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: eta-boot, directory, eta-meta,
template-haskell, aeson, Eta-Haskell-server
> Task :resolveDependenciesEta FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':resolveDependenciesEta'.
> Nonzero (1) exit code for command [/Users/pabloperezgarcia/.gradle/caches/etlas/1.4.0.0/etlas, --select-eta=0.8.0b2, freeze] with workingDir /Users/pabloperezgarcia/Development/Eta-Haskell-server/build/eta-freeze

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s
2 actionable tasks: 1 executed, 1 up-to-date
Nonzero (1) exit code for command [/Users/pabloperezgarcia/.gradle/caches/etlas/1.4.0.0/etlas, --select-eta=0.8.0b2, freeze] with workingDir /Users/pabloperezgarcia/Development/Eta-Haskell-server/build/eta-freeze
21:26:38: Task execution finished 'run'.
```

Someone know if this dependency is available currently in eta-lang?

**Comment by jneira** [comment]
Hi, i am afraid that aeson needs a patch to work with eta. These patches are automatically downloaded from https://github.com/typelead/eta-hackage/tree/master/patches.
There you can check the patched versions for each package and aeson-1.0.2.1 is not. 
You can either use another patched version or request a patch opening an issue in that project.

**Comment by Jyothsnasrinivas** [comment]
Closing this since it looks resolved!
